src/main/java/pl/piekoszek/gorskimatches/equation/SolutionToQuizzesMapper.java

```diff
@@ -110,22 +85,21 @@ public Map<String, Set<String>> insideEquation(String solution) {
 
         for (int i = 0; i < solution.length(); i++) {
             char numberOrSymbolToBeReplaced = solution.charAt(i);
-            var numbersAndSymbolsToBeTakenFrom = takeOneMatchFromNumberOrSymbol.get(numberOrSymbolToBeReplaced);
+            var numbersAndSymbolsWithTakenMatch = takeOneMatchFromNumberOrSymbol.get(numberOrSymbolToBeReplaced);
 
-            for (Character numberWithTakenMatch : numbersAndSymbolsToBeTakenFrom) {
+            for (Character numberWithTakenMatch : numbersAndSymbolsWithTakenMatch) {
                 String quizWithTakenMatch = characterChanger.changeCharactersInString(solution, i, numberWithTakenMatch);
                 for (int k = 0; k < quizWithTakenMatch.length(); k++) {
                     if (k != i) {
                         char numberOrSymbolToAddMatchTo = quizWithTakenMatch.charAt(k);
-                        var numbersAndSymbolsToBeAddedTo = giveOneMatchFromNumberOrSymbol.get(numberOrSymbolToAddMatchTo);
-                        for (Character matchReceiver : numbersAndSymbolsToBeAddedTo) {
+                        var numbersAndSymbolsWithAddedMatch = giveOneMatchFromNumberOrSymbol.get(numberOrSymbolToAddMatchTo);
+                        for (Character matchReceiver : numbersAndSymbolsWithAddedMatch) {
```

Comment:
also not sure about `matchReceiver` name
and it can be `char` instead of `Character` ;)